### ISSUE TITLE
!!!!!!!!!!!!!!!!!!Dead Lock with log!!!!!!!!!!!!!!!!!!

### ISSUE THREAD

**Issue by pmsrlxe** [bumptech/glide#4244]
first Thread
```
"XXThread" prio=5 tid=68 Blocked
  | group="main" sCount=1 dsCount=0 flags=1 obj=0x14041bd0 self=0xc2d4e600
  | sysTid=25258 nice=0 cgrp=default sched=0/0 handle=0xb937c230
  | state=S schedstat=( 158266032 78824021 454 ) utm=12 stm=3 core=7 HZ=100
  | stack=0xb9279000-0xb927b000 stackSize=1040KB
  | held mutexes=
  at com.bumptech.glide.load.engine.Engine.load(Engine.java:-1)
  - waiting to lock <0x0935b401> (a com.bumptech.glide.load.engine.Engine) held by thread 1
  at com.bumptech.glide.request.SingleRequest.onSizeReady(SingleRequest.java:447)
  - locked <0x0a31a8d7> (a com.bumptech.glide.request.SingleRequest)
  at com.bumptech.glide.request.RequestFutureTarget.getSize(RequestFutureTarget.java:125)
  at com.bumptech.glide.request.SingleRequest.begin(SingleRequest.java:272)
  - locked <0x0a31a8d7> (a com.bumptech.glide.request.SingleRequest)
  at com.bumptech.glide.manager.RequestTracker.runRequest(RequestTracker.java:44)
  at com.bumptech.glide.RequestManager.track(RequestManager.java:641)
  - locked <0x0d9d95e8> (a cn.mucang.android.core.glide.GlideRequests)
  at com.bumptech.glide.RequestBuilder.into(RequestBuilder.java:656)
  at com.bumptech.glide.RequestBuilder.into(RequestBuilder.java:622)
  at com.bumptech.glide.RequestBuilder.submit(RequestBuilder.java:775)
  at com.bumptech.glide.RequestBuilder.submit(RequestBuilder.java:757)
```

Second Thread
```"main" prio=5 tid=1 Blocked
  | group="main" sCount=1 dsCount=0 flags=1 obj=0x75f669a0 self=0xf295ce00
  | sysTid=24696 nice=-10 cgrp=default sched=0/0 handle=0xf2edcdc0
  | state=S schedstat=( 5953758953 604552929 5227 ) utm=536 stm=58 core=7 HZ=100
  | stack=0xff683000-0xff685000 stackSize=8192KB
  | held mutexes=
  at com.bumptech.glide.RequestManager.clear(RequestManager.java:-1)
  - waiting to lock <0x0d9d95e8> (a cn.mucang.android.core.glide.GlideRequests) held by thread 68
  at com.bumptech.glide.RequestBuilder.into(RequestBuilder.java:654)
  at com.bumptech.glide.RequestBuilder.into(RequestBuilder.java:622)
  at com.bumptech.glide.RequestBuilder.into(RequestBuilder.java:613)
  at com.bumptech.glide.load.resource.gif.GifFrameLoader.loadNextFrame(GifFrameLoader.java:226)
  at com.bumptech.glide.load.resource.gif.GifFrameLoader.onFrameReady(GifFrameLoader.java:284)
  at com.bumptech.glide.load.resource.gif.GifFrameLoader.loadNextFrame(GifFrameLoader.java:215)
  at com.bumptech.glide.load.resource.gif.GifFrameLoader.start(GifFrameLoader.java:171)
  at com.bumptech.glide.load.resource.gif.GifFrameLoader.subscribe(GifFrameLoader.java:120)
  at com.bumptech.glide.load.resource.gif.GifDrawable.startRunning(GifDrawable.java:243)
  at com.bumptech.glide.load.resource.gif.GifDrawable.start(GifDrawable.java:225)
  at com.bumptech.glide.request.target.ImageViewTarget.maybeUpdateAnimatable(ImageViewTarget.java:134)
  at com.bumptech.glide.request.target.ImageViewTarget.setResourceInternal(ImageViewTarget.java:128)
  at com.bumptech.glide.request.target.ImageViewTarget.onResourceReady(ImageViewTarget.java:104)
  at com.bumptech.glide.request.SingleRequest.onResourceReady(SingleRequest.java:582)
  - locked <0x0d069018> (a com.bumptech.glide.request.SingleRequest)
  at com.bumptech.glide.request.SingleRequest.onResourceReady(SingleRequest.java:544)
  - locked <0x0d069018> (a com.bumptech.glide.request.SingleRequest)
  at com.bumptech.glide.load.engine.Engine.load(Engine.java:185)
  - locked <0x0935b401> (a com.bumptech.glide.load.engine.Engine)
  at com.bumptech.glide.request.SingleRequest.onSizeReady(SingleRequest.java:447)
  - locked <0x0d069018> (a com.bumptech.glide.request.SingleRequest)
  at com.bumptech.glide.request.target.ViewTarget$SizeDeterminer.getSize(ViewTarget.java:389)
  at com.bumptech.glide.request.target.ViewTarget.getSize(ViewTarget.java:221)
  at com.bumptech.glide.request.SingleRequest.begin(SingleRequest.java:272)
  - locked <0x0d069018> (a com.bumptech.glide.request.SingleRequest)
  at com.bumptech.glide.manager.RequestTracker.runRequest(RequestTracker.java:44)
  at com.bumptech.glide.RequestManager.track(RequestManager.java:641)
  - locked <0x0111185d> (a cn.mucang.android.core.glide.GlideRequests)
  at com.bumptech.glide.RequestBuilder.into(RequestBuilder.java:656)
  at com.bumptech.glide.RequestBuilder.into(RequestBuilder.java:716)```

First thread locked 'RequestManager' and then wait for 'Engine' locked by second thread
but second thread waiting for 'RequestManager''s lock

**Comment by pmsrlxe** [comment]
no one response...

**Comment by byecode** [comment]
I have meeting a same issue , is there any way can fix?

**Comment by pmsrlxe** [comment]
> I have meeting a same issue , is there any way can fix?

So helpless...

**Comment by byecode** [comment]
> > I have meeting a same issue , is there any way can fix?
> 
> So helpless...

换fresco吧还是，哈哈哈

**Comment by pmsrlxe** [comment]
> > > I have meeting a same issue , is there any way can fix?
> > 
> > 
> > So helpless...
> 
> 换fresco吧还是，哈哈哈

哈哈哈哈，老铁稳稳的

**Comment by sjudd** [comment]
What version of Glide is this?

**Comment by byecode** [comment]
> What version of Glide is this?

4.9.0

**Comment by pmsrlxe** [comment]
> What version of Glide is this?

also com.github.bumptech.glide:glide:4.9.0

**Comment by sjudd** [comment]
Ok I think this is obsolete, there were a bunch of concurrency fixes in 4.10, and one more in 4.11 (see the release notes).

**Comment by byecode** [comment]
> Ok I think this is obsolete, there were a bunch of concurrency fixes in 4.10, and one more in 4.11 (see the release notes).

tks~
